fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalSubQueryAlias.java

```diff
@@ -156,4 +169,16 @@ public LogicalSubQueryAlias<CHILD_TYPE> withLogicalProperties(Optional<LogicalPr
         return new LogicalSubQueryAlias<>(qualifier, columnAliases, Optional.empty(),
                 logicalProperties, child());
     }
+
+    public int cteId() {
+        return CTEIdGenerator.newCTEId();
```

Comment:
u should add a new function `newCteId` in `StatementScopeIdGenerator`, and generate id from it to avoid overflow.
and do not use primitive int directly, use class `CTEId` instead